src/App.js

```diff
@@ -16,48 +15,91 @@ import Dashboard from './pages/Dashboard';
 import ProjectEdit from './pages/ProjectEdit';
 import TaskDetail from './pages/TaskDetail';
 import ClientDetail from './pages/ClientDetail';
+import './App.css';
 
-function App() {
-  const LoginContainer = () => (
-    <React.Fragment>
-      <Route exact path='/' render={() => <Redirect to='/login' />} />
-      <Route exact path='/login' component={Login} />
-    </React.Fragment>
-  );
-
-  const DefaultContainer = () => (
-    <React.Fragment>
-      <Header />
-      <Switch>
-        <Route exact path='/' component={Dashboard} />
-        <Route path='/dashboard' component={Dashboard} />
-        <Route exact path='/projects' component={Project} />
-        <Route path='/projects/new' component={NewProject} />
-        <Route exact path='/projects/:id' component={ProjectDetail} />
-        <Route exact path='/projects/edit/:id' component={ProjectEdit} />
-        <Route exact path='/clients' component={Clients} />
-        <Route path='/clients/new' component={NewClient} />
-        <Route exact path='/clients/:id' component={ClientDetail} />
-        <Route path='/clients/edit' component={NewClient} />
-        <Route path='/profile' component={ProfilePage} />
-        <Route exact path='/tasks' component={Tasks} />
-        <Route exact path='/tasks/new' component={NewTask} />
-        <Route exact path='/tasks/edit/:id' component={NewTask} />
-        <Route exact path='/tasks/:id' component={TaskDetail} />
-
-        <Route exact path='/administration' component={Users} />
-      </Switch>
-    </React.Fragment>
-  );
-
-  return (
-    <BrowserRouter>
-      <Switch>
-        <Route exact path='/login' component={LoginContainer} />
-        <Route component={DefaultContainer} />
-      </Switch>
-    </BrowserRouter>
-  );
-}
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isAuthenticated: this.isTokenPresent(),
+    };
+
+    this.checkAuthentication = this.checkAuthentication.bind(this);
+    this.isTokenPresent = this.isTokenPresent.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('authenticated', this.checkAuthentication);
+  }
 
-export default App;
+  componentWillUnmount() {
+    document.removeEventListener('authenticated', this.checkAuthentication);
+  }
+
+  checkAuthentication() {
+    this.setState({
+      isAuthenticated: this.isTokenPresent(),
+    });
+  }
+
+  isTokenPresent() {
+    return localStorage.getItem('token') !== null;
+  }
+
+  render() {
+    const LoginContainer = () => (
+      <React.Fragment>
+        <Route exact path='/' render={() => <Redirect to='/login' />} />
```

Comment:
```suggestion
        <Route exact path='/login' component={Login} />
        <Route path='/' render={() => <Redirect to='/login' />} />
```

have you tried this without all other cases, only these two?